compatibility/common/src/main/java/org/projectnessie/tools/compatibility/internal/OldNessie.java

```diff
@@ -37,7 +38,7 @@ private OldNessie() {}
    * This function handles the special cases for Nessie 0.40.0..0.41.0 which have a few issues in
    * the published poms.
    */
-  static ClassLoader oldNessieClassLoader(Version version, String mainArtifactId)
+  static ClassLoader oldNessieClassLoader(Version version, List<String> artifactIds)
```

Comment:
See https://github.com/projectnessie/nessie/pull/6261/files#diff-92429a1562488174d88d4771f53fa1467d2cb03fe26ccdde3ec21ed8812686fbR94 - need to pass more than one "main" artifact to get all required dependencies.